# Osotspa Project EDA

# Import libraries and load all data

In [2]:
# Import libraries
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [22]:
# Load all data
df_customer = pd.read_csv('../ETL/CSV/obd_customer.csv')
df_transaction = pd.read_csv('../ETL/CSV/obd_transaction.csv')
df_visit_plan = pd.read_csv('../ETL/CSV/obd_visit_plan.csv')

C:\Users\User\AppData\Local\Temp\ipykernel_3192\2597314216.py:2: DtypeWarning: Columns (10,11) have mixed types. Specify dtype option on import or set low_memory=False.
  df_customer = pd.read_csv('../ETL/CSV/obd_customer.csv')


# 1. EDA on Customer Table

In [4]:
df_customer.head(10)

dc_code                              dc_name regionname customercode  \
0     LSG  บริษัท แอล เอส กรุ๊ป เทรดดิ้ง จำกัด        USO    LSG003715   
1     LSG  บริษัท แอล เอส กรุ๊ป เทรดดิ้ง จำกัด        USO    LSG003700   
2     LXP                      หลิน เอ็กซ์เพรส        USO    LXP010257   
3     LSG  บริษัท แอล เอส กรุ๊ป เทรดดิ้ง จำกัด        USO    LSG003699   
4     LSG  บริษัท แอล เอส กรุ๊ป เทรดดิ้ง จำกัด        USO    LSG003702   
5     LSG  บริษัท แอล เอส กรุ๊ป เทรดดิ้ง จำกัด        USO    LSG003693   
6     LSG  บริษัท แอล เอส กรุ๊ป เทรดดิ้ง จำกัด        USO    LSG003703   
7     LSG  บริษัท แอล เอส กรุ๊ป เทรดดิ้ง จำกัด        USO    LSG003694   
8     LSG  บริษัท แอล เอส กรุ๊ป เทรดดิ้ง จำกัด        USO    LSG003706   
9     LXP                      หลิน เอ็กซ์เพรส        USO    LXP010255   

                   customername customertype districtname    provincename  \
0                เฮียตี๋ ทางตัน        SUB 2    Undefined         จ.ชุมพร   
1               ศรีสวัสดิ์พานิช        SUB 2    Undefined         จ.ชุมพร   
2                    ป้าสุ(ดาว)        SUB 2    Undefined  จ.สุราษฎร์ธานี   
3                     ธนาทรัพย์        SUB 2    Undefined         จ.ชุมพร   
4  มาบอำมฤตซุปเปอร์ค้าส่ง_ธงฟ้า        SUB 2    Undefined         จ.ชุมพร   
5             วิริยะเจริญทรัพย์        SUB 2    Undefined         จ.ชุมพร   
6                       วุฒิชัย        SUB 2    Undefined         จ.ชุมพร   
7                     ยิ่งเจริญ        SUB 2    Undefined         จ.ชุมพร   
8             วุฒิศักดิ์ค้าข้าว        SUB 1    Undefined         จ.ชุมพร   
9                 ทองพัฒน์พานิช        SUB 2    Undefined  จ.สุราษฎร์ธานี   

  salesmancode latitude longitude ProvinceNameEng  ProvinceID DistrictNameEng  \
0         None     None      None            None        <NA>            None   
1         None     None      None            None        <NA>            None   
2         None     None      None            None        <NA>            None   
3         None     None      None            None        <NA>            None   
4         None     None      None            None        <NA>            None   
5         None     None      None            None        <NA>            None   
6         None     None      None            None        <NA>            None   
7         None     None      None            None        <NA>            None   
8         None     None      None            None        <NA>            None   
9         None     None      None            None        <NA>            None   

   DistrictID  
0        <NA>  
1        <NA>  
2        <NA>  
3        <NA>  
4        <NA>  
5        <NA>  
6        <NA>  
7        <NA>  
8        <NA>  
9        <NA>

In [21]:
df_customer.describe()

Unnamed: 0     ProvinceID     DistrictID
count  910826.000000  909014.000000  909014.000000
mean   455412.500000      39.878182    3997.184443
std    262932.962481      24.201213    2416.650700
min         0.000000      10.000000    1001.000000
25%    227706.250000      16.000000    1606.000000
50%    455412.500000      35.000000    3504.000000
75%    683118.750000      60.000000    6001.000000
max    910825.000000      96.000000    9613.000000

In [5]:
pd.DataFrame(df_customer['customertype'].value_counts()).sort_values(by=['customertype'])

customertype
ร้านค้า O-Zone (ACT)                 1
Survey Outlet                       16
Undefined                           17
Sub Agent 1                         48
CSR Member                         204
Sub Agent 2                        367
SUB 1                              991
SUB 3                             1406
Kiosk/แผงลอย                      1558
โรงงาน                            2916
โรงเรียน(School)                  3186
Gas Station                       6223
ร้านค้าส่งขนาดเล็ก                6859
SUB 2                             7364
อื่นๆ                            15240
ร้าน OFC (โอสถแฟนคลับ)           23294
ร้าน O-Zone                      28129
ร้านอาหาร/ศูนย์อาหาร             28509
Minimart                         32731
โชว์ห่วยใหญ่ (Vinicity)          53434
โชว์ห่วยเล็ก (Community)        698333

In [30]:
pd.DataFrame(df_customer.groupby(['districtname'])['districtname'].count().sort_values(ascending=False))

districtname
districtname                    
อ.เมืองสมุทรปราการ         13657
อ.เมืองนครราชสีมา          10341
อ.บางพลี                    9966
เขตสายไหม                   8339
เขตสวนหลวง                  7013
...                          ...
อ.ทุ่งยางแดง                  20
อ.ไม้แก่น                     20
อ.เกาะพะงัน                   12
อ.เกาะช้าง                     3
อ.เกาะสีชัง                    1

[923 rows x 1 columns]

# 2. EDA on Transaction Table

In [6]:
df_transaction.head(10)

InvoiceNumber InvoiceDate CustomerCode ProductCode        BaseQty  \
0   HSX61C00685  2022-02-01    OWK001831    10000566    6.000000000   
1   HSA22C00334  2022-02-01    RKP003445    10000512  230.000000000   
2   HSR25C00344  2022-02-01    PKA008980    10000548    5.000000000   
3   HSR67C00536  2022-02-01    JTG004975    10000571   24.000000000   
4   HSM61C00018  2022-02-01    PSM000068    10000604   40.000000000   
5   HSU83C00608  2022-02-01    SPC000473    10000378  150.000000000   
6   HSW84C00687  2022-02-01    STB003327    10000399  150.000000000   
7   HSP82C00672  2022-02-01    STW002073    10000650    4.000000000   
8   HSV22C00672  2022-02-01    TPT001299    10000347    6.000000000   
9   HSU64C00535  2022-02-01    TNG000153    10000539    6.000000000   

        NetAmount    DiscAmount     UnitPrice  \
0    46.730000000   1.000000000   8.500000000   
1  2219.630000000  40.000000000  10.500000000   
2    48.600000000          0E-9  10.400000000   
3   272.900000000   4.000000000  12.330000000   
4   500.000000000   5.000000000  13.500000000   
5  1205.610000000  15.000000000  10.500000000   
6  1443.930000000  30.000000000  10.500000000   
7    76.640000000          0E-9  20.500000000   
8    69.160000000          0E-9  12.330000000   
9    69.160000000          0E-9  12.330000000   

                                   ProductNameTH GroupNameLevel1  \
0                 เอ็มเกลือแร่ 250มล เหลือง 1x24   M-ELECTROLYTE   
1                      ลิโพวิตัน-ดี 100มล 1X5X10            LIPO   
2                         ลิโพ-ไฟน์ 100มล 1X5X10            LIPO   
3            คาลพิส โซดา ยูสุ ฮันนี่ 245มล 1X4X6    CALPIS LACTO   
4                   ซี-วิท ทับทิม 140 มล. 1x3x10          C-VITT   
5       เอ็ม-150 วิตามิน บี6 บี12 150 มล. 1x5x10           M-150   
6            ลิโพวิตัน-ดี ฉลากใหม่ 100มล. 1x5x10            LIPO   
7  คารามูโจ้แผ่นเรียบ รสฮ๊อตชิลลี่ 48 กรัม 1x6x4       KARAMUCHO   
8          คาลพิส โซดา องุ่นเคียวโฮ 245 มล 1x4x6    CALPIS LACTO   
9       คาลพิส โซดาออริจินัลน้ำตาลน้อย245มล1X4X6    CALPIS LACTO   

  GroupNameLevel2  
0     SPORT DRINK  
1    ENERGY DRINK  
2    ENERGY DRINK  
3     REFRESHMENT  
4     REFRESHMENT  
5    ENERGY DRINK  
6    ENERGY DRINK  
7            FOOD  
8     REFRESHMENT  
9     REFRESHMENT

In [31]:
df_transaction.dtypes

Unnamed: 0           int64
InvoiceNumber       object
InvoiceDate         object
CustomerCode        object
ProductCode          int64
BaseQty            float64
NetAmount          float64
DiscAmount         float64
UnitPrice          float64
ProductNameTH       object
GroupNameLevel1     object
GroupNameLevel2     object
dtype: object

In [33]:
df_transaction.describe()

Unnamed: 0   ProductCode       BaseQty     NetAmount    DiscAmount  \
count  7.230278e+06  7.230278e+06  7.230278e+06  7.230278e+06  7.230278e+06   
mean   3.615138e+06  1.003765e+07  2.737685e+01  2.625391e+02  3.912233e+00   
std    2.087202e+06  6.079606e+05  8.291878e+01  7.407456e+02  1.878152e+01   
min    0.000000e+00  1.000018e+07  1.000000e+00  5.840000e+00  0.000000e+00   
25%    1.807569e+06  1.000051e+07  1.000000e+01  7.944000e+01  0.000000e+00   
50%    3.615138e+06  1.000060e+07  1.000000e+01  1.246100e+02  0.000000e+00   
75%    5.422708e+06  1.000060e+07  2.000000e+01  1.962600e+02  2.000000e+00   
max    7.230277e+06  2.000173e+07  8.005000e+04  7.705748e+05  1.601000e+04   

          UnitPrice  
count  7.230278e+06  
mean   1.128590e+01  
std    3.622939e+00  
min    6.250000e+00  
25%    8.600000e+00  
50%    1.050000e+01  
75%    1.350000e+01  
max    1.260000e+02

# Find unique values for GroupNameLevel1 and GroupNameLevel2

## GroupNameLevel1 Analysis

In [16]:
# These are the top level-product
pd.DataFrame({'GroupNameLevel1':df_transaction['GroupNameLevel1'].unique()}).set_index('GroupNameLevel1')
pd.DataFrame

Empty DataFrame
Columns: []
Index: [M-ELECTROLYTE, LIPO, CALPIS LACTO, C-VITT, M-150, KARAMUCHO, CHALARM, SOM IN SUM, 12PLUS COS, YANHEE, PEPTEIN, PEPTEIN PLUS, BBM, LIPO PLUS, M-STORM, SHARK COOL BITE, 12PLUS EXCL COS, SLIMMA]

## What is the transaction for each month / product

In [35]:
df_transaction['InvoiceDate'] = pd.to_datetime(df_transaction['InvoiceDate'])
df_transaction.groupby(df_transaction['InvoiceDate'].dt.strftime('%B'))['NetAmount'].sum()

InvoiceDate
April       2.847236e+08
August      3.340409e+07
February    3.800395e+08
July        3.068040e+08
June        3.042403e+08
March       2.790804e+08
May         3.099387e+08
Name: NetAmount, dtype: float64

In [43]:
grouped_products = pd.DataFrame(df_transaction.groupby('GroupNameLevel1').sum()['NetAmount'].sort_values(ascending=False))
grouped_products['percent'] = (df_transaction.groupby('GroupNameLevel1').sum()['NetAmount'] / df_transaction['NetAmount'].sum()) * 100
grouped_products

NetAmount    percent
GroupNameLevel1                         
M-150            8.773375e+08  46.218698
C-VITT           3.671619e+08  19.342322
LIPO             3.183310e+08  16.769880
SOM IN SUM       1.113180e+08   5.864302
CHALARM          1.032051e+08   5.436912
M-ELECTROLYTE    3.852411e+07   2.029475
CALPIS LACTO     3.786260e+07   1.994626
YANHEE           2.692723e+07   1.418543
KARAMUCHO        1.059332e+07   0.558063
PEPTEIN          3.319642e+06   0.174881
PEPTEIN PLUS     1.336536e+06   0.070410
12PLUS COS       9.030594e+05   0.047574
BBM              6.288977e+05   0.033131
LIPO PLUS        5.113349e+05   0.026937
M-STORM          2.222996e+05   0.011711
12PLUS EXCL COS  3.757424e+04   0.001979
SLIMMA           9.364420e+03   0.000493
SHARK COOL BITE  1.181320e+03   0.000062

Top 5 products based on net amount:
1. M150
2. C-VITT
3. LIPO
4. Som in Sum
5. Chalarm

Clearly there's a outlier product, M-150 taking 46.2% of majority of profit

## GroupNameLevel2 - Analysis

In [21]:
pd.DataFrame({'GroupNameLevel2':df_transaction['GroupNameLevel2'].unique()}).set_index('GroupNameLevel2')

Empty DataFrame
Columns: []
Index: [SPORT DRINK, ENERGY DRINK, REFRESHMENT, FOOD, DEODORANT, FUNCTIONAL DRINK, BABY TALCUM, COOL TALCUM]

In [ ]:
pd.DataFrame({'GroupNameLevel2':df_transaction['GroupNameLevel2'].unique()}).set_index('GroupNameLevel2')